packages/mdx/mdx-hast-to-jsx.js

```diff
@@ -168,9 +168,13 @@ MDXContent.isMDXComponent = true`
   if (node.type === 'text') {
     // Don't wrap newlines unless specifically instructed to by the flag,
     // to avoid issues like React warnings caused by text nodes in tables.
-    if (node.value === '\n' && !preserveNewlines) {
+    const shouldPreserveNewlines =
+      preserveNewlines || parentNode.tagName === 'p'
```

Comment:
Hmm, there are other nodes that accept [“phrasing” content](https://github.com/syntax-tree/mdast), but aren’t. Headings and list-items as well, so would it be maybe be needed to add their HTML equivalents here as well?

Reply:
If so, there should be tests for that as well!

Reply:
Good idea, done ✅